pinot-common/src/main/java/org/apache/pinot/common/assignment/InstancePartitions.java

```diff
@@ -134,6 +134,10 @@ public InstancePartitions withName(String newName) {
     return new InstancePartitions(newName, getPartitionToInstancesMap());
   }
 
+  public InstancePartitions createCopy() {
```

Comment:
This is not needed (see the above comment). Also, this is creating a shallow copy, which actually shares the underlying map

Reply:
Ah that was a bad oversight. Thanks for catching this. Fixed.